src/astro/sql/parsers/sql_directory_parser.py

```diff
@@ -0,0 +1,115 @@
+import os
+import re
+from collections import defaultdict
+from typing import Dict
+
+import frontmatter
+from airflow.decorators.base import get_unique_task_id
+from airflow.decorators.task_group import task_group
+from airflow.exceptions import AirflowException
+from airflow.models.xcom_arg import XComArg
+
+from astro.sql.operators.sql_decorator import SqlDecoratoratedOperator
+from astro.sql.table import Table
+
+
+@task_group()
+def render_directory(path, **kwargs):
+    # raise AirflowException(f"Failed because cwd is {os.listdir(path)}, {os.}")
+    files = [
+        f
+        for f in os.listdir(path)
+        if os.path.isfile(os.path.join(path, f)) and f.endswith(".sql")
+    ]
+    template_dict = {}
+
+    # add kwargs to the dictionary if it's something our tables can inherit.
+    for k, v in kwargs.items():
+        if type(v) == XComArg:
+            template_dict[k] = v
+        elif type(v) == Table:
+            template_dict[k] = v
+
+    # Parse all of the SQL files in this directory
```

Comment:
For each block of this function that is commented, it may be worth having smaller functions - which we could potentially cover more easily with unit tests.